modules/core/04-channel/keeper/upgrade_test.go

```diff
@@ -129,6 +130,245 @@ func (suite *KeeperTestSuite) TestChanUpgradeInit() {
 	}
 }
 
+func (suite *KeeperTestSuite) TestChanUpgradeTry() {
+	var (
+		path *ibctesting.Path
+		// expSequence            uint64
+		counterpartyUpgrade    types.Upgrade
+		proposedConnectionHops []string
+	)
+
+	testCases := []struct {
+		name     string
+		malleate func()
+		expPass  bool
+	}{
+		{
+			"success",
+			func() {},
+			true,
+		},
+		// {
+		// 	"success: counterparty upgrade sequence",
+		// 	func() {
+		// 		channel := path.EndpointA.GetChannel()
+		// 		channel.UpgradeSequence = 5
+		// 		path.EndpointA.SetChannel(channel)
+
+		// 		expSequence = 5
+		// 	},
+		// 	true,
+		// },
+		// {
+		// 	"error receipt set with smaller counterparty upgrade sequence",
+		// 	func() {
+		// 		counterpartyUpgradeSequence = 2
+
+		// 		channel := path.EndpointB.GetChannel()
+		// 		channel.UpgradeSequence = 4
+		// 		path.EndpointB.SetChannel(channel)
+		// 	},
+		// 	false,
+		// },
+		{
+			"channel not found",
+			func() {
+				path.EndpointB.ChannelID = ibctesting.InvalidID
+			},
+			false,
+		},
+		{
+			"channel state is not in OPEN or INITUPGRADE state",
+			func() {
+				suite.Require().NoError(path.EndpointB.SetChannelState(types.CLOSED))
+			},
+			false,
+		},
+		{
+			"timeout has passed",
+			func() {
+				counterpartyUpgrade.Timeout = types.NewTimeout(clienttypes.NewHeight(0, 1), 0)
+			},
+			false,
+		},
+		{
+			"invalid connection state",
+			func() {
+				connectionEnd := path.EndpointB.GetConnection()
+				connectionEnd.State = connectiontypes.UNINITIALIZED
+				suite.chainB.GetSimApp().GetIBCKeeper().ConnectionKeeper.SetConnection(suite.chainB.GetContext(), proposedConnectionHops[0], connectionEnd)
```

Comment:
I think this should be:
```suggestion
				suite.chainB.GetSimApp().GetIBCKeeper().ConnectionKeeper.SetConnection(suite.chainB.GetContext(),path.EndpointB.ConnectionID, connectionEnd)
```
If the proposedConnectionID is changed such that it is different from the current connection ID then this test case would not hit the expected code coverage anymore